### PR TITLE
Zarr notebook

### LINKED ISSUE #143: Include Zarr example/notebook
For users getting started, it would be nice to have either an example in a tutorial or a notebook showing how to plug KvikIO into Zarr.

### PR DESCRIPTION
**Pull request by madsbk**
Fixes #143

cc. @ivirshup @jakirkham

### CODE REVIEW

**Comment by jakirkham** [comment]
Thanks Mads! 🙏

Isaac do you want to give this a try? Would appreciate hearing any feedback you have 🙂

**Review by ivirshup — COMMENTED** [review]
Gave this a look, and I think it's great. Thanks for putting this together @madsbk!

One content addition that would be really good is a "CPU compression / GPU decompression" example like in `nvcomp_batch_codec.ipynb`. It would be good for there to be a clear example of how to work with interoperable data, ideally in as "blessed" of a way as possible while https://github.com/zarr-developers/numcodecs/issues/316 is resolved. 

Maybe an intermediate kerchunk store could work? I haven't actually been 100% sure if kerchunk can be used with direct GPU loading (things like https://github.com/rapidsai/kvikio/issues/235 make me think I can, but maybe it's legate specific?), so an example of that could also be great.

I would maybe also consider making this a little more visible. Linking in the readme is good, but I would put documentation about usage above the instructions for a dev install.

**Comment by madsbk** [comment]
Added an example of reading a GPU-written Zarr file into a NumPy array. However, AFAICT it isn't possible to specify a `compressor` when reading a Zarr array thus it will use `kvikio.zarr.Snappy` always.

**Comment by ivirshup** [comment]
Thanks!

> However, AFAICT it isn't possible to specify a compressor when reading a Zarr array thus it will use kvikio.zarr.Snappy always.

In [`nvcomp_batch_codec.ipynb`](https://github.com/rapidsai/kvikio/blob/branch-23.10/notebooks/nvcomp_batch_codec.ipynb), it looks like the `zarr.LZ4` compressor is swapped out with the `kvikio.zarr.LZ4` compressor by overwriting the `.zarray` file in the store. I assume the reverse would work, but it would be nice if we could avoid modifying the original store inplace.

I think it's important to show if data can be read both direct to GPU by this library, and in an environment without the NVIDIA libraries or a GPU.

**Comment by madsbk** [comment]
@Alexey-Kamenev, could I get you to review the notebook. It makes use of your new `NvCompBatchCodec`.

File: notebooks/zarr.ipynb [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+    "z = zarr.open_array(store)\n",
+    "type(z[:]), type(z.compressor), z[:]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8ea73705",
+   "metadata": {},
+   "source": [
+    "### CPU compression / GPU compression\n",
```
[the remark this answers is not in the document]
Comment (madsbk) [reply notebooks/zarr.ipynb:254]: Fixed

File: notebooks/zarr.ipynb [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+   "source": [
+    "z[:] + 42"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7797155f-40f4-4c50-b704-2356ca64cba3",
+   "metadata": {},
+   "source": [
+    "### GPU compression / CPU compression"
```
[the remark this answers is not in the document]
Comment (madsbk) [reply notebooks/zarr.ipynb:143]: Fixed

File: notebooks/zarr.ipynb [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+    "# Let's create a new Zarr array using KvikIO's GDS store and LZ4 compression\n",
+    "z = zarr.array(\n",
+    "    cupy.arange(10), \n",
+    "    chunks=2, \n",
+    "    store=kvikio.zarr.GDSStore(\"my-zarr-file.zarr\"), \n",
+    "    meta_array=cupy.empty(()),\n",
+    "    compressor=NvCompBatchCodec(\"lz4\"),\n",
+    "    overwrite=True,\n",
+    ")\n",
+    "z, z.compressor, z.store"
```
[the remark this answers is not in the document]
Comment (madsbk) [reply notebooks/zarr.ipynb:64]: Yes, I think that is a good idea. Let's do it in a follow up PR

**Comment by madsbk** [comment]
Thanks @Alexey-Kamenev

**Review by quasiben — APPROVED** [review]

**Pull request merged**